KIPs/kip-113.md

```diff
@@ -0,0 +1,216 @@
+
+#### Chain Configuration
+
+In the ChainConfig, the following fields are introduced. All node operators in a network must update `genesis.json` configuration with the same value. The configuration values for Baobab and Cypress networks are hard-coded on the client source code.
+
+- kip113CompatibleBlock: the block number which Klaytn node starts using Registry
+- kip113ContractAddress: the address of Registry
+
+#### Validation
+
+After reading from Registry, the node must perform the following validations:
+
+- [KeyValidate](https://www.ietf.org/archive/id/draft-irtf-cfrg-bls-signature-05.html#section-2.5)
+- [PopVerify](https://www.ietf.org/archive/id/draft-irtf-cfrg-bls-signature-05.html#section-3.3.3)
+
+## Rationale
+
+### Design decisions
+
+#### Registry is not upgradeable
+
+Registry should not be implemented via upgradeable proxy contract due to its impact on the ecosystem:
+
+- Upgrade can be done covertly by the contract owner
+- Upgrade can happen at an arbitrary block
+- Upgrade key can be stolen
+
+#### Propagating the Registry address via chainConfig
+
+There are several methods to propagate the a system contract address to the network:
+
+1. headerGov: deploy the contract from EOA, and make a governance vote from the governing node (the method used for Baobab GovParam address)
+2. contractGov: deploy the contract from EOA, and make an on-chain governance vote
+3. spareContract: deploy the contract from EOA, and set the address to `AddressBook.spareContract`
+4. chainConfig: deploy the contract from EOA, and set the address to chainConfig (the method used for TreasuryRebalance, a.k.a. KIP-103)
+5. precompiled contract: deploy the contract from EOA, and introduce a new precompiled contract that returns its address
+6. bytecode injection: inject a bytecode to a predefined address at hardfork
+
+We can view each method from the following point of view:
+
+##### How to update the address
+
+| Methods                                               | [1] headerGov | [2] contractGov                        | [3] spareContract | [4] chainConfig | [5] precompiled contract | [6] bytecode injection |
+| ----------------------------------------------------- | ------------- | -------------------------------------- | ----------------- | --------------- | ------------------------ | ---------------------- |
+| Address change without hardfork                       | O             | O                                      | O                 | X               | X                        | X                      |
+| Managing address change timing                        | O             | O                                      | X                 | -               | -                        | -                      |
+| Awareness of other contracts after address is updated | X             | △<br> Latest GovParam address required | O                 | -               | -                        | -                      |
+
```

Comment:
It seems like we decided to use chainConfig. Can you explain why? When I see this table, chainConfig is not the best option.

Reply:
Updated the table and added rationale in 5282741. PTAL